.github/workflows/docker-publish.yml

```diff
@@ -1,88 +1,42 @@
 name: Docker
 
-# This workflow uses actions that are not certified by GitHub.
-# They are provided by a third-party and are governed by
-# separate terms of service, privacy policy, and support
-# documentation.
-
 on:
   push:
-    branches: [ main  ]
-    # Publish semver tags as releases.
-    tags: [ 'v*.*.*' ]
+  pull_request:
```

Comment:
Do we want to run this workflow unconditionally on `push` and `pull_request` events? Every push to a feature branch with a PR open will trigger the build twice (first one for `push` and second one for `pull_request.synchronize`). I suppose these files are a part of your generic workflow collection but I wonder if we could be a bit more frugal with the number of actions here.

Reply:
I guess we can copy-paste the thing from the pre-commit workflow:
```
    # Run on external PRs and pushes to branches on the repo
    # This is to avoid double running on PRs from internal branches
    if:
      github.event_name == 'push' || github.event.pull_request.head.repo.full_name !=
      github.repository
```

Reply:
Sounds good – let's just keep in mind that `rust.yml` also runs on the `cron` event and the condition should take this into account.

Reply:
done